hashes/enigma_machine.py

```diff
@@ -40,7 +48,22 @@ def engine(input_character):
     rotator()
 
 
-if __name__ == "__main__":
+def encode_or_decode(message, token):
+    """
+
+    >>> encode_or_decode("hello", 3)
+    (['/', '0', "'", '%', ' '], 3)
+
+    """
+
+    for _ in range(token):
+        rotator()
+    for j in message:
+        engine(j)
+    return code, token
+
+
+def menu():
```

Comment:
This code chunk is all driver code, so it's fine to keep it in the `__main__` block.